tests/replace-add-needed.sh

```diff
@@ -0,0 +1,34 @@
+#! /bin/sh -e
```

Comment:
@Mic92 on master this test actually demonstrates the failure of the issue I believe. Doing the multiple invocations just produces a wrong & failing binary.
```bash

[nix-shell:~/code/github.com/NixOS/patchelf/tests]$ ldd scratch/replace-add-needed/simple 
scratch/replace-add-needed/simple: error while loading shared libraries: /: cannot read file data: Error 21

[nix-shell:~/code/github.com/NixOS/patchelf/tests]$ patchelf --print-needed scratch/replace-add-needed/simple 
/home/fmzakari/code/github.com/NixOS/patchelf/tests/scratch/replace-add-needed/libfoo.so
/home/fmzakari/code/github.com/NixOS/patchelf/tests/scratch/replace-add-needed/libfoo.so
```